Hackney.Shared.Finance/Boundary/Response/ResponseWrapper.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Hackney.Shared.Finance.Boundary.Response
+{
+    public class ResponseWrapper<T> where T : class
```

Comment:
The name of wrapper doesn't reflect well the fact that there **may be** no result.

Obviously, I'm pretty biased towards the `Maybe<T>` as I thought it made method signatures pretty transparent with the fact that they may return no data.

If there's hesitation about the `Maybe` name, perhaps, we can come up with an **alternative name** that represents the duality of the response state?

Reply:
Like `MaybeEmpty<T>` or `PerhapsEmpty<T>` or `PossiblyEmpty<T>`?
Perhaps, you have more ideas?

Reply:
Personally I think it's fine. "Maybe" and "Perhaps" match the usecase now but in the future, we could be extending this class to check for other conditions like, for example, IsChild or something. In which case, keeping the name generic seems to be the correct way to go